From 7ea01a6bc04fe278f16f3016a9f268f454708c9e Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 29 Oct 2019 16:50:34 +0200
Subject: [PATCH 698/741] sdk_dpaa: ls1043a errata: re-enable SG support

commit 227478336ed9acf521ee5aff2764489f49a7d717 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

With certain limitation, SG frames can be used safely without triggering
the errata.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
index a5b6c95311f5..3c3154420167 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
@@ -779,16 +779,6 @@ static int dpa_private_netdev_init(struct net_device *net_dev)
 	/* Advertise NETIF_F_HW_ACCEL_MQ to avoid Tx timeout warnings */
 	net_dev->features |= NETIF_F_HW_ACCEL_MQ;
 
-#ifndef CONFIG_PPC
-	/* Due to the A010022 FMan errata, we can not use S/G frames. We need
-	 * to stop advertising S/G and GSO support.
-	 */
-	if (unlikely(dpaa_errata_a010022)) {
-		net_dev->hw_features &= ~NETIF_F_SG;
-		net_dev->features &= ~NETIF_F_GSO;
-	}
-#endif
-
 	return dpa_netdev_init(net_dev, mac_addr, tx_timeout);
 }
 
-- 
2.17.1

